lib/LaTeXML/Post/LaTeXImages.pm

```diff
@@ -253,7 +253,7 @@ sub generateImages {
       next if -f pathname_absolute($1, $destdir); }
     push(@pending, $table{$key}); }
 
-  NoteProgress(" [$nuniq unique; " . scalar(@pending) . " new]");
+  NoteStatus(2, "LaTeXImages: $nuniq unique; " . scalar(@pending) . " new");
```

Comment:
Looking around I see most `NoteProgress` got turned into `NoteStatus(1` , while this one got a 2 variant. So it's either upgraded or a typo, likely just upgraded. Got me a bit confused, but I'm now assuming you intentionally upgraded it.

There is probably something to be said that there is a better way to name the routine. `Progress` is a bit more topical/intuitive than `Status`, and one may even argue that the word `Note` is not adding much, especially seeing how we have `Debug`, `Info`, `Warning`, `Error`, `Fatal`, each of which could have been  `NoteDebug`, `NoteInfo`, `NoteWarning`, `NoteError`, `NoteFatal`.

So maybe I can even see how calling these variants as `Progress`, `ProgressDetailed`, `ProgressBrief`, etc, may be a pleasing fit with the other inhabitants of Error.pm

Reply:
The number is at what verbosity level to report. I found need for levels 0,1,2, perhaps no others? If so, giving explicit names might be an option.  Progress is indeed more intuitive than Status.  But Progress is also a good name for the now completely different spinner commands; I really don't want to name it Spinner!
Probably needs a name better reflecting processing stages (begin, something, end), but I'm `Stage` doesn't quite capture the sense either.

Reply:
Aren't the spinner prints meant to be a variation on NoteStatus which is interactive, and conditional on the output stream pointing to the actual STDERR?  I assumed they're an internal variation on the same command.

If you want to add them to the external API on the same level as a `Progress` command, a `ProgressInteractive` variation would be reasonable for me. Or even `ProgressSpinner`, which actually starts getting close to name paradigms you may encounter in ReactJS and similar frontend frameworks.

Reply:
An alternative way to say it, is that I mostly (modulo the questions I've listed above) I'm happy with the way the logging & spinners behave. The API naming no longer matches what they do, however. (and perhaps some calls to Note should change wording, level, and so on).
You're right in a way that a `NoteBegin,NodeProgress*,NoteEnd` group is somewhat like a single NoteStatus, and that they all relate to "progress" in some way.  That almost suggests: `NoteStatus`=>`Progress`, `NoteBegin|NoteEnd`=>`BeginProgressGroup|EndProgressGroup` and finally `NoteProgress`=>`ProgressStep`. But only almost :> (and avoiding the `$level` argument by more names just obscures things further).

Reply:
These sound like very nice names to me, almost finished! I again forgot what `NoteStatus(2,` was supposed to be since my last comment, I expect I would need to look it up often.
`Progress`, `ProgressStep`, `ProgressSpinUp`, `ProgressSpinDown`, etc could be fun and memorable, unlike memorizing `2`.

To quote the [constants best practice](https://www.oreilly.com/library/view/perl-best-practices/0596001738/ch04.html#constants):
> Raw numbers that suddenly appear in the middle of a program are often mysterious, frequently confusing, and always a potential source of errors. Certain types of unprintable character strings—for example, initialization strings for modems—are similarly awkward. [...] Replace these kinds of raw literals with a read-only lexical variable whose name explains the meaning of the number

That phrases my sentiment much better than I could.

Reply:
Hmm, your `Progress` set could work. A little odd, but fun.

I don't get the "memorizing 2" bit.  How about `ProgressIfVerbosityIsGreaterThanOrEqualToTwo` ?
Perhaps slightly more seriously, `NoteStatus(1,...)=>Progress(...)`, `NoteStatus(2,...)=>ProgressDetailed(...)`.  Arguably `NoteStatus(0...)` is a special case anyway? I don't know what it should be called, though.

Reply:
Oh, but note that the begin/end things *do* print to the log file, only the step one doesn't; that makes spin up/down still a bit of misnomer.